File: Python/test_suite.py
import unittest
from Problem1 import p1
from Problem2 import p2
from Problem3 import p3
from Problem4 import p4


class MultipleTestCase(unittest.TestCase):
    """
    Test case for checking multiple sum of range method
    """

    print " Euler P1: Running Test Cases for Multiples of 3 and 5"

    def test_for_invalid_values(self):
        self.assertEqual(p1.find_multiples_sum_for_range('Hello'), 0)
        self.assertEqual(p1.find_multiples_sum_for_range(10.00), 0)

    def test_for_invalid_integer(self):
        self.assertEqual(p1.find_multiples_sum_for_range(2), 0)
        self.assertEqual(p1.find_multiples_sum_for_range(-100), 0)

    def test_for_valid(self):
        self.assertEqual(p1.find_multiples_sum_for_range(10), 23)
        self.assertEqual(p1.find_multiples_sum_for_range(6), 8)

class EvenFibonacciTestCase(unittest.TestCase):
    """
    Test case for checking the sum of even numbers in Fibo series
    """

    print " Euler P2: Running Test Cases for Fibonacci Series"
    def test_for_invalid_values(self):
        self.assertEqual(p2.sum_of_even_fibonacci('Hello'), 0)
        self.assertEqual(p2.sum_of_even_fibonacci(10.00), 0)
    
    def test_for_invalid_integer(self):
        self.assertEqual(p2.sum_of_even_fibonacci(-2), 0)
        self.assertEqual(p2.sum_of_even_fibonacci(1), 0)
        self.assertEqual(p2.sum_of_even_fibonacci(2), 0)

    def test_for_valid(self):
        self.assertEqual(p2.sum_of_even_fibonacci(10), 10)
        self.assertEqual(p2.sum_of_even_fibonacci(6), 2)


class PrimeFactorTestCase(unittest.TestCase):
    """
    Test case for checking the greatest prime factor
    """

    print " Euler P3: Running Test Cases for Greatest Prime Factor"

    def test_for_invalid_values(self):
        self.assertEqual(p3.get_greatest_prime_factor('Hello'), 0)
        self.assertEqual(p3.get_greatest_prime_factor(10.00), 0)

    def test_for_invalid_integer(self):
        self.assertEqual(p3.get_greatest_prime_factor(-2), 0)
        self.assertEqual(p3.get_greatest_prime_factor(-40), 0)
        self.assertEqual(p3.get_greatest_prime_factor(1), 0)

    def test_for_valid(self):
        self.assertEqual(p3.get_greatest_prime_factor(10), 5)
        self.assertEqual(p3.get_greatest_prime_factor(6), 3)


class Palindrome6Digit(unittest.TestCase):
    """
    Test case for checking the 6 digit palindrome number
    """

    print " Euler P4: Running Test Cases for 6 Digit Palindrome number"

    def test_for_invalid_values(self):
        self.assertEqual(p4.get_palindrome('Hello'), 0)
        self.assertEqual(p4.get_palindrome(10.00), 0)

    def test_for_invalid_integer(self):
        self.assertEqual(p4.get_palindrome(-2), 0)
        self.assertEqual(p4.get_palindrome(-40), 0)
        self.assertEqual(p4.get_palindrome(1), 0)

    def test_for_valid(self):
        self.assertEqual(p4.get_palindrome(999999), 906609)
        self.assertEqual(p4.get_palindrome(800000), 793397)
        self.assertEqual(p4.get_palindrome(793397), 793397)


if __name__ == '__main__':
    unittest.main()

File: Javascript/test-suite.js
'use strict'

const expect = require('chai').expect;
const findMultiplesOfThreeAndFive = require('./Problem1/p1').findMultiplesOfThreeAndFive;
const findSumOfEvenFibonacci = require('./Problem2/p2').findSumOfEvenFibonacci;
const findGreatestPrimeFactor = require('./Problem3/p3').findGreatestPrimeFactor;
const getGreatestPalindromeForRange = require('./Problem4/p4').getGreatestPalindromeForRange;
const getSmallestMultiple = require('./Problem5/p5').getSmallestMultiple;


describe('Project Euler Problem 1 - Sum of Multiple of 3 and 5', ()=>{

    describe('Invalid test cases', ()=>{
        it('should return 0 for String', ()=>{
            expect(findMultiplesOfThreeAndFive('Grab')).to.equal(0);
        })

        it('should return 0 for Non Integer', () => {
            expect(findMultiplesOfThreeAndFive('-10.9')).to.equal(0);
            expect(findMultiplesOfThreeAndFive('10.9')).to.equal(0);
        })

        it('should return 0 for Invalid Integer', () => {
            expect(findMultiplesOfThreeAndFive(-10)).to.equal(0);
            expect(findMultiplesOfThreeAndFive(3)).to.equal(0);
        })
    })

    describe('Valid test Cases', ()=>{
        it('should return proper output for Valid Integer', () => {
            expect(findMultiplesOfThreeAndFive(4)).to.equal(3);
            expect(findMultiplesOfThreeAndFive(10)).to.equal(23);
        })
    })

});

describe('Project Euler Problem 2 - Sum of even Fibonacci Series number', () => {

    describe('Invalid test cases', () => {
        it('should return 0 for String', () => {
            expect(findSumOfEvenFibonacci('Grab')).to.equal(0);
        })

        it('should return 0 for Non Integer', () => {
            expect(findSumOfEvenFibonacci('-10.9')).to.equal(0);
            expect(findSumOfEvenFibonacci('10.9')).to.equal(0);
        })

        it('should return 0 for Invalid Integer', () => {
            expect(findSumOfEvenFibonacci(-10)).to.equal(0);
            expect(findSumOfEvenFibonacci(1)).to.equal(0);
        })

    })

    describe('Valid test Cases', () => {
        it('should return proper output for Valid Integer', () => {
            expect(findSumOfEvenFibonacci(4)).to.equal(2);
            expect(findSumOfEvenFibonacci(10)).to.equal(10);
            expect(findSumOfEvenFibonacci(40)).to.equal(44);
        })
    })

});

describe('Project Euler Problem 3 - Find greatest Prime factor', () => {

    describe('Invalid test cases', () => {
        it('should return 0 for String', () => {
            expect(findGreatestPrimeFactor('Grab')).to.equal(0);
        })

        it('should return 0 for Non Integer', () => {
            expect(findGreatestPrimeFactor('-10.9')).to.equal(0);
            expect(findGreatestPrimeFactor('10.9')).to.equal(0);
        })

        it('should return 0 for Invalid Integer', () => {
            expect(findGreatestPrimeFactor(-10)).to.equal(0);
            expect(findGreatestPrimeFactor(1)).to.equal(0);
        })

    })

    describe('Valid test Cases', () => {
        it('should return proper output for Valid Integer', () => {
            expect(findGreatestPrimeFactor(4)).to.equal(2);
            expect(findGreatestPrimeFactor(10)).to.equal(5);
            expect(findGreatestPrimeFactor(56)).to.equal(7);
        })
    })

});

describe('Project Euler Problem 4 - Find Palindrome for range', () => {

    describe('Invalid test cases', () => {
        it('should return 0 for String', () => {
            expect(getGreatestPalindromeForRange('Grab')).to.equal(0);
        })

        it('should return 0 for Non Integer', () => {
            expect(getGreatestPalindromeForRange('-10.9')).to.equal(0);
            expect(getGreatestPalindromeForRange('10.9')).to.equal(0);
        })

        it('should return 0 for Invalid Integer', () => {
            expect(getGreatestPalindromeForRange(-10)).to.equal(0);
            expect(getGreatestPalindromeForRange(1)).to.equal(0);
        })

    })

    describe('Valid test Cases', () => {
        it('should return proper output for Valid Integer', () => {
            expect(getGreatestPalindromeForRange(999999)).to.equal(906609);
            expect(getGreatestPalindromeForRange(800000)).to.equal(793397);
            expect(getGreatestPalindromeForRange(793397)).to.equal(793397);
        })
    })

});

describe('Project Euler Problem 5 - Get Smallest Multiple for Range', () => {

    describe('Invalid test cases', () => {
        it('should return 0 for String', () => {
            expect(getSmallestMultiple('Grab')).to.equal(0);
        })

        it('should return 0 for Non Integer', () => {
            expect(getSmallestMultiple('-10.9')).to.equal(0);
            expect(getSmallestMultiple('10.9')).to.equal(0);
        })

        it('should return 0 for Invalid Integer', () => {
            expect(getSmallestMultiple(-10)).to.equal(0);
            expect(getSmallestMultiple(1)).to.equal(0);
        })

    })

    describe('Valid test Cases', () => {
        it('should return proper output for Valid Integer', () => {
            expect(getSmallestMultiple(10)).to.equal(2520);
            expect(getSmallestMultiple(5)).to.equal(60);
            expect(getSmallestMultiple((20))).to.equal(232792560);
        })
    })

});